### PR TITLE
Fix go pointer to go pointer issue

### PR DESCRIPTION
**Pull request by id-ms**

### CODE REVIEW

File: falcon.go [hunk 1]
```diff
@@ -193,7 +189,7 @@ func (sig *CTSignature) SaltVersion() byte {
 //
 // Returns an error if pubkey is invalid.
 func (pub *PublicKey) Coefficients() (h [N]uint16, err error) {
-	r := C.falcon_det1024_pubkey_coeffs((*C.uint16_t)(&h[0]), unsafe.Pointer(pub))
+	r := C.falcon_det1024_pubkey_coeffs((*C.uint16_t)(&h[0]), unsafe.Pointer(&(*pub)))
```
Comment (algorandskiy) [inline falcon.go:192]: what is a point of defererensing and then taking an address again?

Comment (algoidan) [reply falcon.go:192]: it is a common way to access slices and array on cgo.
we use this on curve25519.go
and it is also mention here
https://pkg.go.dev/unsafe#Pointer

Comment (algorandskiy) [reply falcon.go:192]: I'm asking about `pub` that is not an array

Comment (algoidan) [reply falcon.go:192]: `pub` is defined as instance of type `PublicKey`which is defined as
`type PublicKey [PublicKeySize]byte`

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**